cpp/ps0005/cpp_ps0005.cpp

```diff
@@ -58,30 +30,34 @@ int main()
             break;
 
       case '/':
-            if(!num2) {
+            if(num1 && !num2) {
                   std::cout << "Math Error: cannot divide by 0." << std::endl;
-                  return 0;
+                  return 2;
+            }
+            if(!num1 && !num2) {
+                  std::cout << "Invalid character/s or zero valueInfinite number of solutions: (0 / 0) is Undefined Quantity!" << std::endl;
+                  return 3;
             }
             result = num1 / num2;
             break;
 
       case '%':
-            if(std::floor(num1) != num1 || std::floor(num2) != num2) {
+            if(std::int(num1) != num1 || std::int(num2) != num2) {
                   std::cout << "cannot execute modulo operation for floating point numbers." << std::endl;
-                  return 0;
+                  return 4;
             }
             result = static_cast<int>(num1) % static_cast<int>(num2);
             break;
       
       default:
             std::cout << "Invalid character for operator." << std::endl;
-            return 0;
-            break;
+            return 1;
       }

```

Comment:
- [x] used `switch` and `if` structures in a right way to handle different cases
- [x] handled division operation `/`
- [x] handled modulo operation `%`
- [x] handled invalid operator case
- [x] chose the right operation for each case